File: aoc2020/day2/part1.py
def main():
    with open("./input.txt", "r") as f:
        p = f.read().split("\n")

    countValidPass = 0

    for item in p:
        policy, passw = item.split(":")
        count, char = policy.split(" ")
        lowC, upperC = count.split("-")

        charcount = 0
        for c in passw:
            if c == char:
                charcount += 1

        if charcount >= int(lowC) and charcount <= int(upperC):
            countValidPass += 1

    print(countValidPass)

if __name__ == "__main__":
    main()

File: aoc2016/src/day7/part1.js
import * as fs from "fs";

function main() {
  const data = fs
    .readFileSync("./input.txt", "utf-8")
    .split("\n")
    .map((item) => String(item));

  let count = 0;
  const regex = /^(\w)(\w)\2\1$/;
  data.forEach((ip) => {
    if (regex.test(ip)) {
      count += 1;
    }
  });
  console.log(count);
}

main();
